--- a/dmenu_path	2025-08-09 18:00:49.950267502 +0500
+++ b/dmenu_path	2025-10-22 22:17:51.410915496 +0500
@@ -5,9 +5,33 @@
 
 [ ! -e "$cachedir" ] && mkdir -p "$cachedir"
 
+all_commands=$(awk '
+	/^name=/ { alias = substr($0, index($0, "=") + 1); }
+	/^cmds=/ {
+		cmd = substr($0, index($0, "=") + 1);
+		if (alias == "") {
+			print cmd
+		} else {
+			print cmd " & " alias
+		}
+		alias = ""
+	}
+' commands.conf)
+
+echo "$all_commands"
+
 IFS=:
 if stest -dqr -n "$cache" $PATH; then
 	stest -flx $PATH | sort -u | tee "$cache"
+	new=$(cat "$cache")
+
+	while IFS= read -r item
+	do
+		if [[ $new != *"$item" ]]; then
+			echo $item | tee -a "$cache"
+		fi
+	done <<< "$all_commands"
+
 else
-	cat "$cache"
+	cat $cache
 fi
